.github/workflows/linux_meson.yml

```diff
@@ -198,3 +200,55 @@ jobs:
           cd doc
           python3-dbg -m pip install pytest pytest-xdist pytest-timeout mpmath gmpy2 threadpoolctl pooch
           python3-dbg -m pytest --pyargs scipy -n2 --durations=10 -m "not slow"
+
+  #################################################################################
+  gcc8:
+    # Purpose is to examine builds with oldest-supported gcc.
+    name: Build with gcc-8
+    if: "github.repository == 'scipy/scipy' || github.repository == ''"
+    runs-on: ubuntu-20.04  # 22.04 doesn't support gcc-8
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+
+      - name: Setup OS
+        run: |
+          sudo apt-get -y update
+          sudo apt install -y g++-8 gcc-8 gfortran-8
+          sudo apt install -y libatlas-base-dev liblapack-dev libgmp-dev \
+            libmpfr-dev libmpc-dev pkg-config libsuitesparse-dev
+
+          # install openblas
+          chmod +x tools/wheels/cibw_before_build_linux.sh
+          sudo tools/wheels/cibw_before_build_linux.sh .
+
+      - name: Setup Python deps
+        run: |
+          pip install --upgrade "numpy==1.21.6" &&
+          pip install --upgrade pip setuptools==59.6.0 wheel build meson meson-python ninja &&
+          pip install "cython<3.0b1" gmpy2 threadpoolctl mpmath pooch pythran pybind11 pytest pytest-xdist==2.5.0 pytest-timeout
+
+      - name: Build wheel and install
+        run: |
+          set -euo pipefail
+          export PYTHONOPTIMIZE=2
+
+          # specify which compilers to use using environment variables
+          CC=gcc-8 CXX=g++-8 FC=gfortran-8 python -m build --no-isolation
+          pip install dist/scipy*.whl
+
+      - name: Run tests
+        run: |
+          export PYTHONOPTIMIZE=2
+          set -xeuo pipefail
+
+          # can't be in source directory
+          pushd $RUNNER_TEMP
+          bash $GITHUB_WORKSPACE/tools/wheels/cibw_test_command.sh $GITHUB_WORKSPACE
+          popd
```

Comment:
I'd like to avoid this - much more verbose and harder to read than invoking `pytest --pyargs scipy` directly. How about copying exactly what the `python_debug` test stage does? Also the `pipefail` isn't needed then, it can simply be:
```
export PYTHONOPTIMIZE=2
pytest --pyargs scipy -n2
```